### PR TITLE
Loading pattern guidelines

### PR DESCRIPTION
**Pull request by mperrotti**
These guidelines were initiated by the [graceful degradation epic](https://github.com/github/primer/issues/2406). However, these guidelines broadly cover how we should handle loading states - not just how we should handle them in a gracefully degraded experience.

### CODE REVIEW

File: content/ui-patterns/loading.mdx
Comment (mperrotti) [inline content/ui-patterns/loading.mdx:391]: I'm going to get more guidance from @ichelsea when she gets back.

I don't want this to block the PR from being reviewed and merged, so I'll comment this out if we're ready to merge before @ichelsea is back.

File: content/ui-patterns/loading.mdx
Comment (mperrotti) [inline content/ui-patterns/loading.mdx:95]: Reviewers - what do we think?

**Comment by mperrotti** [comment]
Sorry for the long list of commits...I originally branched this off of `mp/graceful-degradation-guidelines`

I'm rebasing to make this easier to review, it's just taking a long time...

File: content/ui-patterns/loading.mdx
Comment (langermank) [reply content/ui-patterns/loading.mdx:95]: I feel like I've only seen this in special cases, or maybe rare cases? I mean you did literally state that 😆 so maybe its okay to include it, so long as we don't end up with a million dancing Mona's.

File: content/ui-patterns/loading.mdx
Comment (langermank) [inline content/ui-patterns/loading.mdx:117]:
```suggestion
The following is meant as general guidance since we can't accurately predict how long a process will take. It's difficult to give even a rough estimate of how long a process will take due to unknown variables such as network speed, device performance, and server load.
```

File: content/ui-patterns/loading.mdx
Comment (langermank) [inline content/ui-patterns/loading.mdx:188]: This video actually shows a scenario I was thinking we might want to cover. There have been a few accessibility violations about loading buttons used within a dialog on submit, where submitting the form loads a new page. In most cases we actually just removed the loading button since the new page loading indicates a state change. 

So not really related to this content section, but wanted to point it out. Although the video is of production so I don't expect you to change it 😆 also, the staff bar is visible in this video FYI.

File: content/ui-patterns/loading.mdx
Comment (langermank) [inline content/ui-patterns/loading.mdx:305]: I might be reading this incorrectly, but I think this section contradicts this section above:

> If non-sequential groups of navigation links are in a loading state, preserve the group headings and replace the contents of each group with a single loading indicator.

Since its showing several loading items next to each other rather than a single loader?

File: content/ui-patterns/loading.mdx
Comment (langermank) [inline content/ui-patterns/loading.mdx:355]: Should this say "semantically disabled" or maybe some way to explain that it visually looks disabled/loading, but is still focusable.

**Review by langermank — COMMENTED** [review]
This is so good 😍 left a few nitpicks and questions!

File: content/ui-patterns/loading.mdx
Comment (mperrotti) [reply content/ui-patterns/loading.mdx:188]: Haha oops! Let me re-record without the staff bar 😅

File: content/ui-patterns/loading.mdx
Comment (mperrotti) [reply content/ui-patterns/loading.mdx:305]: I'm suggesting that we show a single loader for sequential _groups_, not sequential _items_. I totally see the confusion though, and I'll make it more clear.

**Review by langermank — APPROVED** [review]
Looks good, I would just fix that one staff bar video before merging ✨

**Pull request merged**
